### DISCUSSION TITLE
建议将2.0版本源码库分离开来（插件也建议和核心库分离，方便阅读和贡献代码），整个仓库fork下来非常大，访问github本来就慢，下载得好久好久

### DISCUSSION THREAD

**Discussion by lijing666** [antvis/X6#2415] [Q&A]
建议将2.0版本源码库分离开来（插件也建议和核心库分离，方便阅读和贡献代码），整个仓库fork下来非常大，访问github本来就慢，下载得好久好久

**Comment by NewByVector** [comment]
将代码库分开我们也在考虑，插件和核心代码分离其实在[规划](https://www.yuque.com/antv/x6/sk3zdn_nsgg4p)中。👍

**Reply by lijing666** [reply]
恩，看了规划了，比较期待新版本，另外，我看x6的代码，各个模块都好似没有单元测试用例，对于你们核心维护者和开源贡献者来说，如何保证提交代码的可靠性呢？2.0版本是否在考虑这个呢？我本来想给@antv/x6-vue-shape库进行升级，但是一个是拉github仓库太大，一个是如何保证修改后保证可用（单元测试），所以看看是不是等2.0版本我再参与进去，哈哈

**Reply by NewByVector** [reply]
只是为了 clone 小的话，用 --depth=1 只获取最新的commit即可，希望在 2.0 版本中看到你的身影 🥳🥳🥳。单元测试其实现在是我们的一个痛点，因为没有时间来做这个事情，而且这个非常重要，导致我们修改代码如履薄冰。

**Reply by lijing666** [reply]
好的，我先升级一下v1版本的@antv/x6-vue-shape的vue-demi，旧版强制依赖于@vue/composition-api是不对的，vue-demi官方已经修复

**Reply by lijing666** [reply]
就升级了vue-demi为latest版本，直接提交了pr，但是整个工程不知道怎么测试，有点尴尬

**Reply by zmcode** [reply]
> 只是为了 clone 小的话，用 --depth=1 只获取最新的commit即可，希望在 2.0 版本中看到你的身影 🥳🥳🥳。单元测试其实现在是我们的一个痛点，因为没有时间来做这个事情，而且这个非常重要，导致我们修改代码如履薄冰。

2.0考虑使用canvas来渲染吗?

**Reply by NewByVector** [reply]
@yaojin2070 暂时不会的，我们追求的是节点易定制、开箱即用，svg 目前阶段能帮我解决很多问题。如果要使用 canvas，可以尝试 G6.

**Reply by zmcode** [reply]
> @yaojin2070 暂时不会的，我们追求的是节点易定制、开箱即用，svg 目前阶段能帮我解决很多问题。如果要使用 canvas，可以尝试 G6.

我看了一些issue, 对于svg使用foreignObject来嵌入html, ie11是不是不支持?  还有就是对于上千节点的数据是不是不推荐使用x6?

**Reply by NewByVector** [reply]
对 后续我们会完全放弃 ie11， 对于节点过多的场景，我们在 x6 2.0 中做了大量优化，能覆盖一些高性能场景，如果节点 > 2000 以上的话，推荐使用 G6。

**Reply by zmcode** [reply]
> 对 后续我们会完全放弃 ie11， 对于节点过多的场景，我们在 x6 2.0 中做了大量优化，能覆盖一些高性能场景，如果节点 > 2000 以上的话，推荐使用 G6。

你们说完全放弃IE11, 如果真的要兼容IE11的时候, 我们可以自己通过webpack去打包一个umd来实现吗?不是很懂webpack配置和babel, 就以目前1.3的版本来说

**Reply by NewByVector** [reply]
就是新的功能特性不会在 IE11 上测试，在 IE 上不保证正确性。
